src/extractor/guidance/intersection_handler.cpp

```diff
@@ -400,108 +393,105 @@ std::size_t IntersectionHandler::findObviousTurn(const EdgeID via_edge,
 
     const EdgeData &in_way = node_based_graph.GetEdgeData(via_edge);
 
-    std::vector<out_way> out_ways;
-
-    // create sortable vector of only entry allowed out_ways
     // get a count of number of ways from that intersection that qualify to have
     // continue instruction because they share a name with the approaching way
     std::int64_t continue_count = 0, continue_count_valid = 0;
     for (std::size_t i = 1, last = intersection.size(); i < last; ++i)
     {
-        out_way out;
-        const auto out_node = node_based_graph.GetEdgeData(intersection[i].eid);
+        const auto out_data = node_based_graph.GetEdgeData(intersection[i].eid);
 
-        out.road = &intersection[i];
-        out.same_name_id = !util::guidance::requiresNameAnnounced(
-            in_way.name_id, out_node.name_id, name_table, street_name_suffix_table);
-        if (out.same_name_id)
+        bool out_same_name_id = !util::guidance::requiresNameAnnounced(
```

Comment:
can be put into a lambda `const auto is_continue_road = [&in_way,this](const ConnectedRoad &road){ const auto &out_data = ...; return !requires...; }`

After you can call `std::count_if(intersection.begin(),intersection.end(),is_continue_road)` to count the continues.

The `count valid` part can probably be moved to an additional check in `intersection` (just as findClosestTurn) since it can be used at other parts of the code as well.